tests/test_ess_csc.py

```diff
@@ -94,27 +98,72 @@ async def test_bin_script(self):
         await self.check_bin_script(name="ESS", index=1, exe_name="run_ess_csc.py")
 
     async def validate_telemetry(self):
+        mtt = common.MockTestTools()
         for sensor_name in self.csc.device_configurations:
-            device_configuration = self.csc.device_configurations[sensor_name]
-            data = await self.remote.tel_temperature.next(flush=False)
-            assert data.numChannels == device_configuration.num_channels
-            temperature = data.temperature
-            temperature = data.temperature[: device_configuration.num_channels]
-            assert common.MockTemperatureConfig.min <= min(temperature)
-            assert common.MockTemperatureConfig.max >= max(temperature)
+            md_props = common.MockDeviceProperties(name=sensor_name)
+            device_config: common.DeviceConfig = self.csc.device_configurations[
+                sensor_name
+            ]
+            if device_config.sens_type == common.SensorType.TEMPERATURE:
+                md_props.num_channels = device_config.num_channels
+                data = await self.remote.tel_temperature.next(flush=False)
+
+                # First make sure that the temperature data contain the
+                # expected number of NaN values.
+                expected_num_nans = len(data.temperature) - device_config.num_channels
+                nan_array = [math.nan] * expected_num_nans
+                np.testing.assert_array_equal(
+                    nan_array, data.temperature[device_config.num_channels :]
+                )
+
+                # Next validate the rest of the data.
+                assert data.numChannels == device_config.num_channels
+                reply = [
+                    data.sensorName,
+                    data.timestamp,
+                    common.ResponseCode.OK,
+                ] + data.temperature[: device_config.num_channels]
+                mtt.check_temperature_reply(md_props=md_props, reply=reply)
```

Comment:
We also talked privately about the md_props argument. I agree with you it would probably be better to use named arguments to the check_x_reply functions, instead of a struct-like class. Please consider that for a future cleanup ticket.

Reply:
I'll address this in DM-32273.